crates/biome_migrate/src/rule_mover.rs

```diff
@@ -427,56 +438,82 @@ impl AnalyzerMover {
             }
         }
 
+        for _ in 0..linter_members.len().saturating_sub(1) {
+            linter_separators.push(token(T![,]));
+        }
+
+        for _ in 0..assist_members.len().saturating_sub(1) {
+            assist_separators.push(token(T![,]));
+        }
+
         let list = self
             .root
             .value()
             .ok()?
             .as_json_object_value()?
             .json_member_list();
 
-        let mut members: Vec<_> = list
-            .iter()
-            .filter_map(|el| {
-                let el = el.ok()?;
-                let token_text = el.name().ok()?.inner_string_text().ok()?;
-                if token_text.text() == "linter"
-                    || token_text.text() == "assist"
-                    || self.filters.iter().any(|s| s.as_ref() == token_text.text())
-                {
-                    None
-                } else {
-                    Some(el)
+        let mut separators = vec![];
+        let mut top_level_members = vec![];
+
+        // AKAK: linter: {}
+        let linter_member = self
+            .linter_member
+            .unwrap_or(create_new_linter_member(vec![], vec![]));
+
+        // AKAK: assist: {}
+        let assist_memebr = self
+            .assist_member
+            .unwrap_or(create_new_assist_member(vec![], vec![]));
+
+        for top_level_member in list.iter().flatten() {
+            let name = top_level_member
+                .name()
+                .ok()
+                .and_then(|n| n.inner_string_text().ok());
+
+            if let Some(name) = name {
+                let should_filter = self
+                    .filters
+                    .iter()
+                    .any(|filter| name.text() == filter.as_ref());
+
+                if should_filter {
+                    continue;
                 }
-            })
-            .collect();
-        let mut separators: Vec<_> = list.separators().filter_map(|el| el.ok()).collect();
 
-        if !linter_members.is_empty() {
-            for _ in 0..linter_members.len() - 1 {
-                linter_separators.push(token(T![,]))
-            }
-
-            let new_linter_member = create_new_linter_member(linter_members, linter_separators);
-            members.push(new_linter_member);
-            if members.len() > 1 {
-                separators.push(token(T![,]));
+                if name.text() == "linter" {
+                    if linter_members.is_empty() {
+                        top_level_members.push(top_level_member.clone());
+                    } else {
+                        let new_linter_member = add_members_to_linter_member(
+                            &linter_member,
+                            linter_members.clone(),
+                            linter_separators.clone(),
+                        );
```

Comment:
The first implementation was adding the new member at the very end. While it's more efficient, it basically moves around an existing member, causing plenty of diffs. 

I decided that it's best to pay a bit more in memory, but at least we do swap the members in the same position, so the number the final diff is very limited.